File: src/com/anathema_roguelike/characters/perks/targetingstrategies/constraints/LineOfSight.java
package com.anathema_roguelike.characters.perks.targetingstrategies.constraints;

import com.anathema_roguelike.characters.Character;
import com.anathema_roguelike.characters.perks.targetingstrategies.Targetable;

public class LineOfSight<T extends Targetable> implements TargetConstraint<T, Character> {

	@Override
	public boolean apply(T target, Character character) {
		return character.hasLineOfSightTo(target);
	}
}

File: src/com/anathema_roguelike/entities/characters/perks/actions/targetingstrategies/constraints/LineOfSight.scala
package com.anathema_roguelike
package entities.characters.perks.actions.targetingstrategies.constraints

import com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.Targetable
import com.anathema_roguelike.entities.characters.Character

class LineOfSight[T <: Targetable] extends TargetConstraint[T, Character] {
  override def apply(target: T, character: Character): Boolean = character.hasLineOfSightTo(target)
}